src/cow-react/modules/limitOrders/updaters/QuoteUpdater/hooks/useGetMarketPrice.ts

```diff
@@ -0,0 +1,99 @@
+import { useAtomValue, useSetAtom } from 'jotai'
+import { useCallback, useEffect } from 'react'
+import { useWeb3React } from '@web3-react/core'
+import { useUpdateAtom } from 'jotai/utils'
+
+import { getQuote } from '@cow/api/gnosisProtocol'
+import { SimpleGetQuoteResponse } from '@cowprotocol/cow-sdk'
+import { adjustDecimals } from '@cow/modules/limitOrders/utils/adjustDecimals'
+import { useLimitOrdersTradeState } from '@cow/modules/limitOrders/hooks/useLimitOrdersTradeState'
+import { limitRateAtom, updateLimitRateAtom } from '@cow/modules/limitOrders/state/limitRateAtom'
+import { limitOrdersQuoteAtom } from '@cow/modules/limitOrders/state/limitOrdersQuoteAtom'
+import { useQuoteRequestParams } from './useQuoteRequestParams'
+
+// Every 10s
+const REFETCH_CHECK_INTERVAL = 10000
+
+export function useGetMarketPrice() {
+  const { chainId, account } = useWeb3React()
+
+  const feeQuoteParams = useQuoteRequestParams()
+  const updateLimitRateState = useUpdateAtom(updateLimitRateAtom)
+  const setLimitOrdersQuote = useSetAtom(limitOrdersQuoteAtom)
+
+  const { inputCurrency, outputCurrency, orderKind } = useLimitOrdersTradeState()
+  const { isInversed } = useAtomValue(limitRateAtom)
+
+  // Handle error
+  const handleError = useCallback(
+    (error) => {
+      updateLimitRateState({ executionRate: null })
+      console.debug('[useFetchMarketPrice] Failed to fetch exection price', error)
+    },
+    [updateLimitRateState]
+  )
+
+  // Handle response
+  const handleResponse = useCallback(
```

Comment:
Would be great to move `handleResponse` from `useGetMarketPrice` to independent hook